### PR TITLE
Improve handling of KeyedService in Transient Disposables detection

### PR DESCRIPTION
**Pull request by nwoolls**
The modification updates the conditional checks inside Transient Disposables service detection to include consideration for KeyedService. It now verifies whether a KeyedService exists, checking both ImplementationType and ImplementationFactory for each service offering, be it keyed or non-keyed.

Fixes https://github.com/dotnet/blazor-samples/issues/282

### CODE REVIEW

**Comment by nwoolls** [comment]
Note that I kept this diff pretty clean without any refactoring. Let me know if you are interested in me introducing some more modern / idiomatic C# as well. I wasn't sure what the balance was of keeping sample code "simple" vs. "modern".

```csharp

switch (descriptor.Lifetime)
{
    case ServiceLifetime.Transient 
        when (descriptor is { IsKeyedService: true, KeyedImplementationType: not null }
              && typeof(IDisposable).IsAssignableFrom(descriptor.KeyedImplementationType))
             || (descriptor is { IsKeyedService: false, ImplementationType: not null }
                 && typeof(IDisposable).IsAssignableFrom(descriptor.ImplementationType)):
        collection.Add(CreatePatchedDescriptor(descriptor));
        break;
    case ServiceLifetime.Transient
        when descriptor is { IsKeyedService: true, KeyedImplementationFactory: not null }
            or { IsKeyedService: false, ImplementationFactory: not null }:
        collection.Add(CreatePatchedFactoryDescriptor(descriptor));
        break;
    default:
        collection.Add(descriptor);
        break;
}
```

**Comment by guardrex** [comment]
Thanks @nwoolls ... I'll get to this soon ... hopefully by EOD but certainly by Friday morning.

**Comment by guardrex** [comment]
Yes, you can apply changes that you feel improve the composition of the code.

File: .gitignore [hunk 1]
```diff
@@ -16,6 +16,7 @@ _site/
 !.vscode/extensions.json
 !.vscode/settings.json
 *.userprefs
+.idea/
```
[the remark this answers is not in the document]
Comment (nwoolls) [reply .gitignore]: Done - added to my global `.gitignore`.

**Comment by guardrex** [comment]
Thanks @nwoolls! ... LGTM 👍 ... but I'm too busy to test it 🙈😄. Normally, I would do so, but I'm buried in other work at the moment. I'll roll with it and take further feedback if anyone has a concern. Thanks again for working this in. It's very helpful. I'll cross-link and close the doc-side issue now.

**Pull request merged**